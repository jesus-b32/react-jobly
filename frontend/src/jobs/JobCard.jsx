import React, { useContext, useState, useEffect } from 'react';
import UserContext from '../auth/UserContext';
import {
  Card,
  CardBody,
  CardTitle,
  CardSubtitle,
  ListGroup,
  ListGroupItem,
  Button,
} from 'reactstrap';

function JobCard({ job }) {
  const { hasAppliedToJob, applyToJob } = useContext(UserContext);
  const [jobApplied, setJobApplied] = useState();

  /** Check if jobs has already been applied to and update state */
  useEffect(() => {
    setJobApplied(hasAppliedToJob(job.id));
  }, [job.id, hasAppliedToJob]);

  /** Apply for a job */
  async function handleJobApply(event) {
    if (hasAppliedToJob(job.id)) return;
    applyToJob(job.id);
    setJobApplied(true);
  }

  return (
    <>
      <Card>
        <CardBody className='text-center'>
          <CardTitle tag={'h4'}>{job.title}</CardTitle>
          {job.companyName ? (
            <CardSubtitle
              tag={'h6'}
            >{`Company: ${job.companyName}`}</CardSubtitle>
          ) : null}
          <ListGroup className='mt-4' flush>
            <ListGroupItem>{`Salary: ${job.salary || 0}`}</ListGroupItem>
            <ListGroupItem>{`Equity: ${job.equity || 0}`}</ListGroupItem>
          </ListGroup>
          {jobApplied ? (
            <Button className='mt-4' color='danger' disabled>
              Applied
            </Button>
          ) : (
            <Button className='mt-4' color='danger' onClick={handleJobApply}>
              Apply
            </Button>
          )}
        </CardBody>
      </Card>
    </>
  );
}

export default JobCard;
