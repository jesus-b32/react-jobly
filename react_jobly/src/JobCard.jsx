import React from 'react';
import {
  Card,
  CardBody,
  CardTitle,
  CardSubtitle,
  ListGroup,
  ListGroupItem,
} from 'reactstrap';

function JobCard({ job }) {
  return (
    <>
      <Card>
        <CardBody className='text-center'>
          <CardTitle tag={'h4'}>{job.title}</CardTitle>
          {job.companyName ? (
            <CardSubtitle tag={'h6'}>job.companyName</CardSubtitle>
          ) : null}
          <ListGroup>
            <ListGroupItem>{job.salary}</ListGroupItem>
            <ListGroupItem>{job.equity}</ListGroupItem>
          </ListGroup>
        </CardBody>
      </Card>
    </>
  );
}

export default JobCard;
