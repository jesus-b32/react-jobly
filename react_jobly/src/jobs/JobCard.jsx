import React from 'react';
import {
  Card,
  CardBody,
  CardTitle,
  CardSubtitle,
  ListGroup,
  ListGroupItem,
  Button,
} from 'reactstrap';

function JobCard({ job }) {
  return (
    <>
      <Card>
        <CardBody className='text-center'>
          <CardTitle tag={'h4'}>{job.title}</CardTitle>
          {job.companyName ? (
            <CardSubtitle
              tag={'h6'}
            >{`Company: ${job.companyName}`}</CardSubtitle>
          ) : null}
          <ListGroup className='mt-4' flush>
            <ListGroupItem>{`Salary: ${job.salary}`}</ListGroupItem>
            <ListGroupItem>{`Equity: ${job.equity}`}</ListGroupItem>
          </ListGroup>
          <Button className='mt-4'>Apply</Button>
        </CardBody>
      </Card>
    </>
  );
}

export default JobCard;
