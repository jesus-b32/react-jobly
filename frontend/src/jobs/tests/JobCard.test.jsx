import JobCard from '../JobCard';
import { MemoryRouter } from 'react-router-dom';
import { jobData } from '../../tests/testCommon';
import { it, expect, vi } from 'vitest';
import { render } from '@testing-library/react';
import UserContext from '../../auth/UserContext';

// const setCurrentUser = vi.fn();
const hasAppliedToJob = vi.fn();
const applyToJob = vi.fn();

const TestJobCard = () => {
  const job = jobData[0];
  return (
    <MemoryRouter>
      <UserContext.Provider value={{ hasAppliedToJob, applyToJob }}>
        <JobCard job={job} />
      </UserContext.Provider>
    </MemoryRouter>
  );
};

it('should render without crashing', () => {
  render(<TestJobCard />);
});

it('should match snapshot', () => {
  const { asFragment } = render(<TestJobCard />);
  expect(asFragment()).toMatchSnapshot();
});
