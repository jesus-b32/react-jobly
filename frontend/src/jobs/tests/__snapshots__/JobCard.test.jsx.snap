// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="mb-4 card"
  >
    <div
      class="text-center card-body"
    >
      <h4
        class="card-title"
      >
        test job 1
      </h4>
      <h6
        class="card-subtitle"
      >
        Company: test company 1
      </h6>
      <ul
        class="mt-4 list-group list-group-flush"
      >
        <li
          class="list-group-item"
        >
          Salary: 100000
        </li>
        <li
          class="list-group-item"
        >
          Equity: 0.3
        </li>
      </ul>
      <button
        class="mt-4 btn btn-danger"
        type="button"
      >
        Apply
      </button>
    </div>
  </div>
</DocumentFragment>
`;
